File: ClassicalAndLondonSchools/ClassicalAndLondonSchools/Customer.cs
namespace ClassicalAndLondonSchools;

public class Customer
{
    /// <summary>
    /// Purchase specific amount of a product from the store.
    /// </summary>
    /// <param name="store">
    /// The store containing products.
    /// </param>
    /// <param name="product">
    /// The product to purchase.
    /// </param>
    /// <param name="amount">
    /// The amount to purchase.
    /// </param>
    /// <returns>
    /// Returns true if successfully purchased the product from the store; Returns false otherwise.
    /// </returns>
    public bool Purchase(IStore store, Products product, int amount)
    {
        bool isPurchased = false;
        if (store.HasEnoughInventory(product, amount))
        {
            store.RemoveInventory(product, amount);
            isPurchased = true;
        }
        return isPurchased;
    }
}

File: classical-and-london-schools-of-testings-java/src/store/Customer.java
package store;

public class Customer {
    /**
     * Purchase specific amount of a product from the store.
     * 
     * @param store   The store containing products.
     * @param product The product to purchase.
     * @param amount  The amount to purchase.
     * @return Returns true if successfully purchased the product from the store;
     *         Returns false otherwise.
     */
    public boolean purchase(Store store, Products product, int amount) {
        boolean isPurchased = false;
        if (store.hasEnoughInventory(product, amount)) {
            store.removeInventory(product, amount);
            isPurchased = true;
        }
        return isPurchased;
    }
}
